soh/soh/Enhancements/game-interactor/GameInteractor.cpp

```diff
@@ -50,6 +50,50 @@ bool GameInteractor::IsGameplayPaused() {
     return (Player_InBlockingCsMode(gPlayState, player) || gPlayState->pauseCtx.state != 0 || gPlayState->msgCtx.msgMode != 0) ? true : false;
 }
 
-bool GameInteractor::CanSpawnEnemy() {
+bool GameInteractor::CanSpawnActor() {
     return GameInteractor::IsSaveLoaded() && !GameInteractor::IsGameplayPaused();
 }
+
+bool GameInteractor::CanAddOrTakeAmmo(int16_t amount, int16_t item) {
+    int16_t upgradeToCheck = 0;
+
+    switch (item) {
+        case ITEM_STICK:
+            upgradeToCheck = UPG_STICKS;
+            break;
+        case ITEM_NUT:
+            upgradeToCheck = UPG_NUTS;
+            break;
+        case ITEM_BOW:
+            upgradeToCheck = UPG_QUIVER;
+            break;
+        case ITEM_SLINGSHOT:
+            upgradeToCheck = UPG_BULLET_BAG;
+            break;
+        case ITEM_BOMB:
+            upgradeToCheck = UPG_BOMB_BAG;
+            break;
+        default:
+            break;
+    }
+
+    if (amount < 0 && AMMO(item) == 0) {
+        return false;
+    }
+
+    if (item != ITEM_BOMBCHU && item != ITEM_BEAN) {
+        if ((CUR_CAPACITY(upgradeToCheck) == 0) || (amount > 0 && AMMO(item) == CUR_CAPACITY(upgradeToCheck))) {
+            return false;
+        }
+        return true;
+    } else {
+        // Seperate checks for beans and bombchus because they don't have capacity upgrades
+        if (INV_CONTENT(item) != item ||
+            (amount > 0 && ((item == ITEM_BOMBCHU && AMMO(item) == 50) || (item == ITEM_BEAN && AMMO(item) == 10)))) {
```

Comment:
Should giving bombchus take into consideration when bombchus are in logic or not, similar to when buying bombchus from a shop?

i.e. can only give bombchus if player has the bomb bag when chus aren't in logic, or has found a progressive chu when chus are in logic.

This is the shop logic for reference: https://github.com/HarbourMasters/Shipwright/blob/877fc2dcca71edcb17b51e9f802a77ceb0f41500/soh/src/overlays/actors/ovl_En_GirlA/z_en_girla.c#L678-L686

Reply:
This effect or GameInteractor function isn't neccesarily tied to rando, and I don't think it neccesarily hurts too much having this be "out of logic" (they can take them away too after all), but I'm open to implementing this if others feel like this should be implemented.

I think for a viewer it might be a bit more confusing though.

Reply:
Decided to leave this simple for now. If we ever get requests to implement this check, we can do it then.